indico/modules/networks/controllers.py

```diff
@@ -0,0 +1,96 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from flask import request, session
+from sqlalchemy.orm import contains_eager
+
+from MaKaC.webinterface.rh.admins import RHAdminBase
+from indico.core.db import db
+from indico.modules.categories.models.categories import Category
+from indico.modules.events.models.events import Event
+from indico.modules.networks import logger
+from indico.modules.networks.forms import IPNetworkGroupForm
+from indico.modules.networks.models.networks import IPNetworkGroup
+from indico.modules.networks.views import WPNetworksAdmin
+from indico.web.util import jsonify_data, jsonify_form, jsonify_template
+
+
+class RHManageNetworks(RHAdminBase):
+    """Management list for IPNetworks"""
+
+    def _process(self):
+        network_groups = IPNetworkGroup.find().order_by(IPNetworkGroup.name).all()
+        return WPNetworksAdmin.render_template('networks.html', network_groups=network_groups)
+
+
+class RHCreateIPNetworkGroup(RHAdminBase):
+    """Dialog to create an IPNetworkGroup"""
+
+    CSRF_ENABLED = True
+
+    def _process(self):
+        form = IPNetworkGroupForm()
+        if form.validate_on_submit():
+            network_group = IPNetworkGroup()
+            form.populate_obj(network_group)
+            db.session.add(network_group)
+            logger.info('Network group "{}" created by {}'.format(network_group.name, session.user))
+            return jsonify_data(flash=False)
+        return jsonify_form(form)
+
+
+class RHAdminIPNetworkGroupBase(RHAdminBase):
+    """Base class for managing in IPNetworkGroup"""
+
+    def _checkParams(self):
+        self.network_group = IPNetworkGroup.find_one(id=request.view_args['network_group_id'])
+
+
+class RHEditIPNetworkGroup(RHAdminIPNetworkGroupBase):
+    """Dialog to edit an IPNetworkGroup"""
+
+    def _process(self):
+        form = IPNetworkGroupForm(obj=self.network_group)
+        if form.validate_on_submit():
+            old_name = self.network_group.name
+            form.populate_obj(self.network_group)
+            logger.info('Network group "{}" edited by {}'.format(old_name, session.user))
+            return jsonify_data(flash=False)
+        return jsonify_form(form)
+
+
+class RHDeleteIPNetworkGroup(RHAdminIPNetworkGroupBase):
+    """Dialog to delete an IPNetworkGroup"""
+
+    def _process_GET(self):
+        query = (self.network_group.in_event_acls
+                 .join(Event)
+                 .order_by(Event.title)
+                 .options(contains_eager('event_new')))
+        events = [principal.event_new for principal in query]
+        query = (self.network_group.in_category_acls
+                 .join(Category)
+                 .order_by(Category.title)
+                 .options(contains_eager('category')))
+        categories = [principal.category for principal in query]
+        in_use = bool(events or categories)
```

Comment:
i'd probably just check `events or categories` in the template, but no strong opinion